### PR TITLE
bugfix/distribute-diagnostic-sheet-generation-per-figure

### PR DESCRIPTION
**Pull request by ritvikvasan**
**Pull request recommendations:**
- [ ] Updated function to use Dask for plotting as well (not just collection paths)

### CODE REVIEW

**Comment by ritvikvasan** [comment]
Here is the data for this in quilt: https://staging.allencellquilt.org/b/allencell-internal-quilt/tree/aics/actk/bugfix.make_diagnostic_sheet/makediagnosticsheet/

File: actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py [hunk 1]
```diff
@@ -71,199 +70,32 @@ def __init__(
...
-            figure_paths.append(
-                dataset.loc[dataset[this_metadata] == fig_num][
-                    DatasetFields.DiagnosticSheetPath
-                ].values
+        # Load feature to plot if feature
+        if feature:
+            with open(row[DatasetFields.CellFeaturesPath]) as f:
+                cell_features = json.load(f)
+            title = "CellId: {0}, {1} {2}: {3}".format(
+                row[DatasetFields.CellId], "\n", feature, cell_features[feature],
```
Comment (JacksonMaxfield) [inline actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py:87]: Can you use an fstring here instead:

```python
f"""CellId: {row[DatasetFields.CellId]
{feature}: {cell_features[feature]}"""
```

File: actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py [hunk 2]
```diff
@@ -403,12 +244,32 @@ def run(
             # Process each row
             for j, this_metadata in enumerate(metadata):
 
+                # Add some helper columns for subsequent analysis
+
+                # "SubplotNumber" + str(this_metadata) + "/MaxCells" is a new column
```
Comment (evamaxfield) [inline actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py]: I think it is somewhat odd to include a `/` in a column name. Can you change it an underscore or something?

Comment (evamaxfield) [reply actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py]: Or ohhhhh I see, its a calculated thing. I would say like `SubplotNumberModCells`?

**Review by evamaxfield — COMMENTED** [review]
Talked with Ritvik on Slack call. More memory and process safe to distribute the generation of a single diagnostic sheet. So basically a function that takes a dataframe and creates a figure and axes during the live time of the function rather than passing the axes of the figure to the distributed function.

Changes to come before merging.

**Comment by ritvikvasan** [comment]
Quilt link: https://staging.allencellquilt.org/b/allencell-internal-quilt/tree/aics/actk/bugfix.make_diagnostic_sheet/makediagnosticsheet/. Plotted many figures here and it seemed fast

**Review by evamaxfield — APPROVED** [review]
Thanks for making all of these changes.

**Pull request merged**